src/dda/build/metadata/enums.py

```diff
@@ -0,0 +1,170 @@
+# SPDX-FileCopyrightText: 2025-present Datadog, Inc. <dev@datadoghq.com>
+#
+# SPDX-License-Identifier: MIT
+
+from enum import StrEnum, auto
+from typing import ClassVar
+
+from msgspec import Struct
+
+
+class ArtifactType(StrEnum):
+    """
+    The type of the build artifact, one of:
+    - `comp` (component)
+    - `dist` (distribution)
+    - `other` for miscellaneous artifacts: source code tarball, configuration files, etc.
+    """
+
+    COMP = auto()
+    DIST = auto()
+    OTHER = auto()
+
+
+class ArtifactFormat(StrEnum):
+    """
+    The format of the build artifact.
+    """
+
+    BIN = auto()
+    SRC = auto()  # Source code tarball
+    DEB = auto()
+    RPM = auto()
+    MSI = auto()
+    CFG = auto()  # Configuration files tarball
+    DOCKER = auto()  # Docker container image
+
+    def validate_for_type(self, artifact_type: ArtifactType) -> None:
```

Comment:
This makes me think that we should rely more on type checking and have something like the concept of variants for each of the high-level types rather than a separate format enumeration.